core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -1206,6 +1219,19 @@ class ParallelReduce<FunctorType, Kokkos::MDRangePolicy<Traits...>, ReducerType,
   using DummySHMEMReductionType = int;
 
  public:
+  template <typename Policy, typename Functor>
+  static int max_tile_size_product(const Policy& pol, const Functor&) {
+    cudaFuncAttributes attr =
+        CudaParallelLaunch<ParallelReduce,
+                           LaunchBounds>::get_cuda_func_attributes();
+    auto const& prop = pol.space().cuda_device_prop();
+    // Limits due do registers/SM
+    int const regs_per_sm        = prop.regsPerMultiprocessor;
+    int const regs_per_thread    = attr.numRegs;
+    int const max_threads_per_sm = regs_per_sm / regs_per_thread;
+    return std::min(max_threads_per_sm,
+                    int(Kokkos::Impl::CudaTraits::MaxHierarchicalParallelism));
```

Comment:
Maybe a static_cast instead of a C-style cast?

Reply:
Good call

Reply:
https://github.com/kokkos/kokkos/pull/3481/commits/a4d0e0178fcb326d88e071373bffd8614ed21258